File: number_theory/least_prime_factor.py
# User function Template for python3
class Solution:
    def leastPrimeFactor(self, n):
        # code here
        ans = [1] * (n + 1)
        for i in range(2, n + 1):
            if ans[i] == 1:
                ans[i] = i
                for j in range(i * i, n + 1, i):
                    if ans[j] == 1:
                        ans[j] = i
        return ans


# {
# Driver Code Starts
# Initial Template for Python 3
if __name__ == "__main__":
    T = int(input())
    for i in range(T):
        n = int(input())
        ob = Solution()
        ans = ob.leastPrimeFactor(n)
        for i in range(1, n + 1):
            print(ans[i], end=" ")
        print()

# } Driver Code Ends

File: number_theory/least_prime_factor.cpp
//{ Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++
class Solution {
public:
  vector<int> leastPrimeFactor(int n) {
    // code here
    vector<int> ans(n + 1, 1);
    for (int i = 2; i <= n; ++i) {
      for (int j = 2; j <= i; ++j) {
        if (i % j == 0) {
          ans[i] = j;
          break;
        }
      }
    }
    return ans;
  }
};

//{ Driver Code Starts.
int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    Solution ob;
    vector<int> ans = ob.leastPrimeFactor(n);
    for (int i = 1; i <= n; i++)
      cout << ans[i] << " ";
    cout << endl;
  }
  return 0;
}

// } Driver Code Ends
